packages/astro/src/cli/install-package.ts

```diff
@@ -161,3 +162,56 @@ async function installPackage(
 		return false;
 	}
 }
+
+export async function fetchPackageJson(
+	scope: string | undefined,
+	name: string,
+	tag: string
+): Promise<Record<string, any> | Error> {
+	const packageName = `${scope ? `${scope}/` : ''}${name}`;
+	const registry = await getRegistry();
+	const res = await fetch(`${registry}/${packageName}/${tag}`);
+	if (res.status >= 200 && res.status < 300) {
+		return await res.json();
+	} else if (res.status === 404) {
+		// 404 means the package doesn't exist, so we don't need an error message here
```

Comment:
This is a quite odd pattern that `astro add` does everywhere where functions return errors, kinda neat